src/Services/request.ts

```diff
@@ -0,0 +1,66 @@
+import { soxa } from "../../deps.ts";
+import {
+  EServiceKindError,
+  GithubErrorResponse,
+  GithubExceedError,
+  QueryDefaultResponse,
+  ServiceError,
+} from "../Types/index.ts";
+
+export async function requestGithubData<T = unknown>(
+  query: string,
+  variables: { [key: string]: string },
+  token = "",
+) {
+  const response = await soxa.post("", {}, {
+    data: { query, variables },
+    headers: {
+      Authorization: `bearer ${token}`,
+    },
+  }) as QueryDefaultResponse<{ user: T }>;
+  const responseData = response.data;
+
+  if (!responseData?.data?.user) {
+    throw handleError(
+      responseData as unknown as GithubErrorResponse | GithubExceedError,
+    );
+  }
+
+  if (responseData.data.user) {
+    return responseData.data.user;
+  }
+
+  throw new ServiceError("not found", EServiceKindError.NOT_FOUND);
```

Comment:
Is there a pattern to reach this process?

Reply:
TBH, i just wish to separate the fetch file, in near future, i want to create a good structure for the project as Next / Fresh does.

Reply:
I understand the vision. But, how would you handle this process?It appears to be a useless process.

Reply:
Good catch, removed the useless if